app/models/course.rb

```diff
@@ -53,6 +53,15 @@ class Course < ActiveRecord::Base
   accepts_nested_attributes_for :invitations, :assessment_categories
 
   scope :ordered_by_title, -> { order(:title) }
+  scope :ordered_by_end_at, ->(direction = :desc) { order(end_at: direction) }
+
+  # @!method with_approved_user
+  #   Selects all the courses with user as one of its approved members
+  scope :with_approved_user, (lambda do |user|
```

Comment:
This name actually a bit confusing to me. how about `of_user` or `from_user` ? @weiqingtoh any comments?